youtube_dl/extractor/sverigesradio.py

```diff
@@ -0,0 +1,107 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..compat import compat_str
+from ..utils import (
+    int_or_none,
+    try_get,
+)
+
+
+class SverigesRadioBaseIE(InfoExtractor):
+    _BASE_URL = 'https://sverigesradio.se/sida/playerajax'
+
+    def _extract_format(self, query, audio_id, audio_type, fatal=True):
+        audiourl = self._download_json(
+            self._BASE_URL + '/getaudiourl', audio_id,
+            fatal=fatal, query=dict(query, type=audio_type, quality='high', format='iis'))
+        if audiourl and ('audioUrl' in audiourl) and audiourl['audioUrl'] != "":
+            return [{'vcodec': 'none', 'url': compat_str(audiourl['audioUrl'])}]
+        else:
+            return []
+
+    def _extract_audio(self, audio_type, secondary_type, url):
+        audio_id = self._match_id(url)
+        query = {'id': audio_id, 'type': audio_type}
+
+        metadata = self._download_json(self._BASE_URL + '/audiometadata', audio_id, query=query)
+        item = metadata['items'][0]
+
+        formats = self._extract_format(query, audio_id, audio_type)
+        if secondary_type is not None:
+            formats += self._extract_format(query, audio_id, secondary_type, False)
+
+        return {
+            'id': audio_id,
+            'formats': formats,
+            'title': compat_str(item['subtitle']),
```

Comment:
no need to call compat_str if it's already as `string` and `title` field should be extracted before formats.